domain/secretbackend/state/state.go

```diff
@@ -81,6 +81,78 @@ WHERE  uuid = $M.uuid`, sqlair.M{}, ModelSecretBackend{})
 	}, nil
 }
 
+// GetModelType returns the model type for the given model UUID.
+func (s *State) GetModelType(ctx context.Context, modelUUID coremodel.UUID) (coremodel.ModelType, error) {
+	db, err := s.DB()
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+
+	msb := ModelSecretBackend{ModelID: modelUUID}
+	stmt, err := s.Prepare(`
+SELECT model_type AS &ModelSecretBackend.model_type
+FROM   v_model_secret_backend
+WHERE  uuid = $ModelSecretBackend.uuid`, msb)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, stmt, msb).Get(&msb)
+		if errors.Is(err, sql.ErrNoRows) {
+			return fmt.Errorf("cannot get model type for model %q: %w", modelUUID, modelerrors.NotFound)
+		}
+		return errors.Trace(err)
+	})
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	return msb.ModelType, nil
+}
+
+// GetInternalAndActiveBackendUUIDs returns the UUIDs for the internal and active secret backends.
+func (s *State) GetInternalAndActiveBackendUUIDs(ctx context.Context, modelUUID coremodel.UUID) (string, string, error) {
+	db, err := s.DB()
+	if err != nil {
+		return "", "", errors.Trace(err)
+	}
+
+	msb := ModelSecretBackend{ModelID: modelUUID}
```

Comment:
Maybe "activeBackend" instead of "msb".
And "internalBackend" instead of "sb".
A bit more clear to the reader.